### ISSUE TITLE
`paraInclusion` is massively overestimating its weight

### ISSUE THREAD

**Issue by ggwpez** [paritytech/polkadot#5192]
Take a look at some recent [Polkadot blocks](https://polkadot.js.org/apps/#/explorer/query/). Even empty blocks consume ~500ms of weight.   
Benchmarking such blocks shows that only a fraction (~4%) of the weight is actually needed.  
That means `paraInclusion` is massively overestimating its weight cost.  
@shawntabrizi noted that we could think about refunding weight here.
![Screenshot from 2022-03-23 15-57-10](https://user-images.githubusercontent.com/10380170/159730516-4f7e20cb-c312-4f0c-a608-fb846f0bc2a1.png)

Output from [benchmark-block](https://github.com/paritytech/substrate/pull/11091) on reference hardware with production profile and wasm executor:
```pre
Block 9556370 with     2 tx used   4.29% of its weight (    21,509,600 of    501,617,735 ns)    
Block 9556371 with     3 tx used   4.08% of its weight (    20,643,956 of    505,412,849 ns)    
Block 9556372 with     2 tx used   3.94% of its weight (    19,750,293 of    501,617,735 ns)    
Block 9556373 with     2 tx used   3.98% of its weight (    20,368,847 of    511,839,008 ns)    
Block 9556374 with     3 tx used   3.97% of its weight (    19,920,982 of    501,891,126 ns)    
Block 9556375 with     3 tx used   3.98% of its weight (    20,114,849 of    505,370,681 ns)    
```

**Comment by shawntabrizi** [comment]
I think probably we can solve this with a weight refund. My guess is this overestimate comes from the fact that we must assume a lot of things about how many parachains there are, how many are upgrading, how many are sending messages, etc...

If we can track those real numbers and stick it back into the benchmark function, then we can refund any unused weight.

**Comment by burdges** [comment]
It depends what weight means too, like does off chain work count?  We only do availability when backing a parachain block, but `paraInclusion` triggers approval checks with like 40-50 nodes downloading and checking the parachain block.  Also approval checks shall become the dominant rewards source for validators.

**Comment by ggwpez** [comment]
I'm not sure if I understand your question. The weight should reflect the time that it takes validators to import the block.  
If you want to specifically incentivize something, you can modify the fee per weight for that.  

Otherwise we have the problem like right now that the blocks are 25% full without much in them. @burdges

**Comment by burdges** [comment]
Alright cool.  We de facto rate limit approvals, etc. by the number of availability cores anyways, which makes sense.

**Comment by rphmeier** [comment]
Yeah @burdges this only concerns the on-chain weight, not the implicit weight of coordinated off-chain stuff.

**Comment by kianenigma** [comment]
I think this should be tackled not too long in the future. Perhaps someone from FRAME team can give you a hand (az Zeke did last year), especially if someone knowledgable is willing to help out.
